front/src/applications/stdcm/components/StdcmResults/StdcmResults.tsx

```diff
@@ -96,93 +96,106 @@ const StdcmResults = ({
   }, [hasSimulationResults, outputs]);
 
   const [similarTrains, setSimilarTrains] = useState<SimilarTrainWithSecondaryCode[]>([]);
+  const [areSegmentsLoading, setAreSegmentsLoading] = useState(false);
```

Comment:
You can set it to true by default and only update it once the similar trains are loaded
```suggestion
  const [areSegmentsLoading, setAreSegmentsLoading] = useState(true);
```

Reply:
Done